src/codewhisperer/util/runtimeLanguageContext.ts

```diff
@@ -3,103 +3,147 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+import { getLogger } from '../../shared/logger/logger'
 import { CodewhispererLanguage } from '../../shared/telemetry/telemetry.gen'
 import { createConstantMap, ConstantMap } from '../../shared/utilities/tsUtils'
 import * as codewhispererClient from '../client/codewhisperer'
 import * as CodeWhispererConstants from '../models/constants'
 
+type NormalizedLanguageId = Exclude<CodewhispererLanguage, 'jsx' | 'tsx'>
+
+const normalizedLanguageSet: ReadonlySet<NormalizedLanguageId> = new Set([
+    'c',
+    'cpp',
+    'csharp',
+    'go',
+    'java',
+    'javascript',
+    'kotlin',
+    'php',
+    'python',
+    'ruby',
+    'rust',
+    'scala',
+    'shell',
+    'sql',
+    'typescript',
+])
+
 export class RuntimeLanguageContext {
     /**
-     * A map storing cwspr supporting programming language with key: vscLanguageId and value: cwsprLanguageId
-     * Key: vscLanguageId
+     * Key: Union set of CodewhispererLanguageId and PlatformLanguageId (VSC, C9 etc.)
      * Value: CodeWhispererLanguageId
      */
-    private supportedLanguageMap: ConstantMap<CodeWhispererConstants.SupportedLanguage, CodewhispererLanguage>
+    private supportedLanguageMap: ConstantMap<
+        CodeWhispererConstants.PlatformLanguageId | CodewhispererLanguage,
+        CodewhispererLanguage
+    >
 
     /**
      * A map storing CodeWhisperer supported programming language with key: vscLanguageId and value: language extension
      * Key: vscLanguageId
      * Value: language extension
      */
-    private supportedLanguageExtensionMap: ConstantMap<CodeWhispererConstants.SupportedLanguage, string>
-
-    // A set contains vscode languageId and CodeWhispererLanguage
-    private supportedLanguageSet = new Set<string>()
+    private supportedLanguageExtensionMap: ConstantMap<CodewhispererLanguage, string>
 
     constructor() {
-        this.supportedLanguageMap = createConstantMap<CodeWhispererConstants.SupportedLanguage, CodewhispererLanguage>({
-            java: 'java',
-            python: 'python',
-            javascriptreact: 'jsx',
-            javascript: 'javascript',
-            typescript: 'typescript',
-            typescriptreact: 'tsx',
-            csharp: 'csharp',
+        this.supportedLanguageMap = createConstantMap<
+            CodeWhispererConstants.PlatformLanguageId | CodewhispererLanguage,
```

Comment:
maybe I'm missing something, is this map for mapping platformId to telemetry id? why do we need `CodewhispererLanguage` in the input type?

Reply:
purpose of this is to make this map, language type agnostic, at least against PlatformLanguage & CodewhispererLanguage. Callers are safe to call as many times as they want without side effect.

Reply:
the status quo  is that we are using both PlatformLanguageId and CodewhispererLanguage anywhere without consistency. Take `CodeCoverageTracker.getTracker` for example, https://github.com/aws/aws-toolkit-vscode/blob/master/src/codewhisperer/service/recommendationHandler.ts#L302 is using `session.language` which has type CodewhispererLanguage whereas in activation.ts https://github.com/aws/aws-toolkit-vscode/blob/master/src/codewhisperer/activation.ts#L310, `e.document.languageId` is being used

Reply:
Making this map to be `PlatformLanguageId | CodewhispererLanguage` type agnostic is one way to reduce the surface of changes we have to make for this purpose

Reply:
The ultimate goal is still to use an unified language interface

Reply:
I see it makes sense now thanks

Reply:
For the keys here, do they also need to include C9 platform languageId?